File: lang/ts-plugins/src/diagram/diagram-plugin.ts
import type { FlowDefinition } from '@unlaxer/tramli';
import { MermaidGenerator } from '@unlaxer/tramli';
import type { DiagramBundle } from './types.js';

export class DiagramPlugin {
  generate<S extends string>(definition: FlowDefinition<S>): DiagramBundle {
    const mermaid = MermaidGenerator.generate(definition);
    const json = definition.dataFlowGraph?.toJson() ?? '{}';
    const md =
      `# ${definition.name}\n\n` +
      `- initial: \`${definition.initialState}\`\n` +
      `- states: \`${definition.allStates().length}\`\n` +
      `- transitions: \`${definition.transitions.length}\`\n`;
    return { mermaid, dataFlowJson: json, markdownSummary: md };
  }
}

File: lang/java-plugins/src/main/java/org/unlaxer/tramli/plugins/diagram/DiagramPlugin.java
package org.unlaxer.tramli.plugins.diagram;

import org.unlaxer.tramli.FlowDefinition;
import org.unlaxer.tramli.FlowState;
import org.unlaxer.tramli.MermaidGenerator;

public final class DiagramPlugin {
    public <S extends Enum<S> & FlowState> DiagramBundle generate(FlowDefinition<S> definition) {
        String mermaid = MermaidGenerator.generate(definition);
        String json = definition.dataFlowGraph().toJson();
        String md = "# " + definition.name() + "\n\n"
                + "- initial: `" + definition.initialState() + "`\n"
                + "- states: `" + definition.allStates().size() + "`\n"
                + "- transitions: `" + definition.transitions().size() + "`\n";
        return new DiagramBundle(mermaid, json, md);
    }
}
